src/game_options.c

```diff
@@ -0,0 +1,130 @@
+#include <nitro.h>
+#include <string.h>
+
+
+#include "constants/game_options.h"
+
+#include "core_sys.h"
+#include "savedata.h"
+#include "heap.h"
+#include "unk_02025E08.h"
+#include "game_options.h"
+#include "core_sys.h"
+
+Options* Options_AllocMemory (u32 heapID)
+{
+    Options *options = Heap_AllocFromHeap(heapID, sizeof(Options));
+    Options_Init(options);
+
+    return options;
+}
+
+void Options_Copy (const Options *src, Options *dest)
+{
+    MI_CpuCopy8(src, dest, sizeof(Options));
+}
+
+void Options_Init (Options *options)
+{
+    MI_CpuFill8(options, 0, sizeof(Options));
+
+    options->textSpeed = OPTIONS_TEXT_SPEED_NORMAL;
+    options->soundMethod = OPTIONS_SOUND_METHOD_STEREO;
+    options->battleStyle = OPTIONS_BATTLE_STYLE_SHIFT;
+    options->battleScene = OPTIONS_BATTLE_SCENE_ON;
+    options->buttonMode = OPTIONS_BUTTON_MODE_NORMAL;
+    options->frame = OPTIONS_FRAME_1;
+}
+
+void Options_SetSystemButtonMode (SaveData *saveData, int buttonMode)
+{
+    if (saveData != NULL) {
+        buttonMode = Options_ButtonMode(sub_02025E44(saveData));
+    }
+
+    switch (buttonMode) {
+    case OPTIONS_BUTTON_MODE_START_IS_X:
+        gCoreSys.buttonMode = BUTTON_MODE_START_IS_X;
+        break;
+    case OPTIONS_BUTTON_MODE_L_IS_A:
+        gCoreSys.buttonMode = BUTTON_MODE_L_IS_A;
+        break;
+    case OPTIONS_BUTTON_MODE_NORMAL:
```

Comment:
Does this `case` branch get optimized out? Does the binary match if you remove it?

Reply:
Nope, it doesn't match without that there.